### PR TITLE
Improve monorepo support for no-extraneous-dependencies rule

### PR DESCRIPTION
**Pull request by hulkish**
Solves #935

### CODE REVIEW

File: src/rules/no-extraneous-dependencies.js [hunk 1]
```diff
@@ -7,22 +7,66 @@ import importType from '../core/importType'
...
+    const packageContent = {
+      dependencies: {},
+      devDependencies: {},
+      optionalDependencies: {},
+      peerDependencies: {},
+    }
 
-    if (!packageContent) {
-      return null
+    if (Object.prototype.hasOwnProperty.call(context.settings, 'import/paths')) {
```
Comment (ljharb) [inline src/rules/no-extraneous-dependencies.js]: let's use the `has` module for this.

**Comment by hulkish** [comment]
@ljharb It's a documented approach in the readme for lerna: https://github.com/lerna/lerna/blob/master/README.md#common-devdependencies

Furthermore, node itself inherently resolves dependencies from parent `node_modules` dirs. 

When you have... let's say 10-30 packages you're managing in a monorepo - then sure, not a big deal to manage dev deps on a per package basis.

However, at my job for example we manage a monorepo will well over 100 packages.

In this scenario; the task of managing dev deps for this volume of packages quickly outgrows what's practical.

While I agree that this is up for debate on a best practices basis - I'm not sure how valueable it is to have this flexibility prevented at the level of an eslint plugin.

Personally, I've found that hoisting all of these dev deps to the root of monorepo brings a significant benefit and gained sanity.

**Comment by tizmagik** [comment]
Thank you for this PR @hulkish -- this has been a pain point for us so far. Hopefully this gets merged and released 🙏

**Comment by hulkish** [comment]
@ljharb hoping this can land.. is there anything I can improve to allow it?

**Comment by godban** [comment]
Thank for PR. We also reach this problem in monorepo. There even extra plugin in the wild https://github.com/Dreamscapes/eslint-import-resolver-lerna which partially solves it.
But have a solution in `eslint-plugin-import` is definitely better.

**Comment by dmk255** [comment]
I've also run into this issue. I would love to see this merged.

**Comment by hulkish** [comment]
@benmosher hmm, sure - but, how would you configure `packageDir` in your `.eslintrc`?

**Comment by dmk255** [comment]
@hulkish I believe the relevant PR is https://github.com/benmosher/eslint-plugin-import/pull/685

**Comment by benmosher** [comment]
thanks for the PR reference @dmk255! other relevant reading: 

- #1053 
- #458

**Comment by evocateur** [comment]
I just turn off `import/no-extraneous-dependencies` for any test files in the monorepo with an [`overrides`](https://eslint.org/docs/user-guide/configuring#configuration-based-on-glob-patterns) key. `import/no-unresolved` already handles "is it installed somewhere", so extraneous-ness in test files seems beside the point. `${ROOT}/.eslintrc.yaml` looks like this:

```yaml
---
  extends: airbnb
  root: true
  overrides:
    - files: "**/*{-,.}{test,stories}.js"
      rules:
        # devDependencies are all in the root
        import/no-extraneous-dependencies: off
```

**Comment by hulkish** [comment]
I'll just fork and publish this as a different package.

**Comment by ljharb** [comment]
You’re welcome to do that, but this should stay open.

**Comment by hulkish** [comment]
Just wondering... what's holding this up? What needs to be done to get this merged?

**Comment by benmosher** [comment]
busy week at work, but it's top of my queue for this project when I get time

**Comment by hulkish** [comment]
@benmosher thanks!

**Comment by benmosher** [comment]
@hulkish fyi, merged mine because I had fixed up and added a few more tests, LMK if I missed anything though

**Pull request closed without merging**